data/rules.json

```diff
@@ -41,7 +41,7 @@
         }
       },
       {
-        "id": "usablePerimeter",
+        "id": "upperbound",
```

Comment:
Should fact IDs be unique in case both perimeter and area rules are added to the same action? e.g. `perimeterUpperBound` and `areaUpperBound`

Reply:
It's a bit of a fudge to get this in - by using 'upperbound' we can then use the fact 'upperbound' output by the rules engine. If an action were to have perimeter and area rules, then what you say would require a change, but for the moment neither action will do this

Reply:
I thought that might be the case. Fair enough, we can stick with a single fact for now.

I guess if we did enable both area and perimeter rules for the same action, we'd run into more issues than just this because it would also need 2 inputs to make any sense.